packages/terra-responsive-element/src/ResponsiveElement.jsx

```diff
@@ -0,0 +1,122 @@
+   */
+  tiny: PropTypes.element,
+  /**
+   * An element to be displayed at small breakpoints
+   */
+  small: PropTypes.element,
+  /**
+   * An element to be displayed at medium breakpoints
+   */
+  medium: PropTypes.element,
+  /**
+   * An element to be displayed at large breakpoints
+   */
+  large: PropTypes.element,
+  /**
+   * An element to be displayed at huge breakpoints
+   */
+  huge: PropTypes.element,
+  /**
+   * The viewport the element will be responsive to, one of `window` or `parent`
+   */
+  responsiveTo: PropTypes.oneOf(['window', 'parent']),
+};
+
+const defaultProps = {
+  responsiveTo: 'parent',
+};
+
+class ResponsiveElement extends React.Component {
+  constructor() {
+    super();
+    this.state = { element: null };
+    this.setContainer = this.setContainer.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.handleWindowResize = this.handleWindowResize.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.container) {
+      this.resizeObserver = new ResizeObserver((entries) => { this.handleResize(entries[0].contentRect.width); });
+      this.resizeObserver.observe(this.container);
+    } else {
+      this.handleResize(window.innerWidth);
+      window.addEventListener('resize', this.handleWindowResize);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.container) {
+      this.resizeObserver.disconnect(this.container);
+      this.container = null;
+    } else {
+      window.removeEventListener('resize', this.handleWindowResize);
+    }
+  }
+
+  setContainer(node) {
+    if (node === null) { return; } // Ref callbacks happen on mount and unmount, element will be null on unmount
+    this.container = this.props.responsiveTo === 'parent' ? node.parentNode : null;
+  }
+
+  handleResize(width) {
+    let element;
+    const { tiny, small, medium, large, huge } = getBreakpoints();
+    if (width >= huge && this.props.huge) {
+      element = this.props.huge;
+    } else if (width >= large && this.props.large) {
+      element = this.props.large;
+    } else if (width >= medium && this.props.medium) {
+      element = this.props.medium;
+    } else if (width >= small && this.props.small) {
+      element = this.props.small;
+    } else if (width >= tiny && this.props.tiny) {
+      element = this.props.tiny;
+    } else {
+      element = this.props.defaultElement;
+    }
+
+    if (this.state.element !== element) {
+      this.setState({ element });
+    }
+  }
+
+  handleWindowResize() {
+    this.handleResize(window.innerWidth);
+  }
+
+  render() {
+    /* eslint-disable no-unused-vars */
+    const { defaultElement, tiny, small, medium, large, huge, responsiveTo, ...customProps } = this.props;
+
+    const className = classNames(
+      'terra-ResponsiveElement',
+      { [`${customProps.className}`]: customProps.className },
+    );
+
+    return (
+      <div {...customProps} ref={this.setContainer} className={className}>
```

Comment:
NIT comment.. is there a known use case for needing to allow custom class names on this element? Wondering if we don't just limit this to responsive element switching.